File: programmers/level1/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    int temp=0;
    vector<int>ath;

    for(int i=0;i<n;i++){
        ath.push_back(1);
    }
    for(int i=0;i<lost.size();i++){
        ath[lost[i]-1]--;
    }
    for(int i=0;i<reserve.size();i++){
        ath[reserve[i]-1]++;
    }

    for(int i=0;i<n;i++){
        if(ath[i]==1){answer++;}
        else if(ath[i]==2){
            answer++;
            if(ath[i-1]==0&&i>0){
                answer++;
            }
            else if(ath[i+1]==0){
                ath[i+1]=1;
            }
        }
    }
    return answer;
}

File: programmers/level1/체육복.js
function solution(n, lost, reserve) {
    let answer = 0;
    let arr = Array.from({length:n+1}, () => 1);
    for(let i of lost)
        arr[i]--;
    for(let i of reserve)
        arr[i]++;
    
    for(let i=1; i<=n; i++){
        if(arr[i] === 2 && arr[i-1] === 0){
            arr[i-1]++;
            arr[i]--;
        }else if(arr[i] === 2 && arr[i+1] === 0){
            arr[i]--;
            arr[i+1]++;
        }
    }
    
    for(let i=1; i<=n; i++)
        if(arr[i])
            answer++;
            
    return answer;
}
